CMakeLists.txt

```diff
@@ -125,7 +125,8 @@ if(${JWT_SSL_LIBRARY} MATCHES "wolfSSL")
   target_link_libraries(jwt-cpp INTERFACE PkgConfig::wolfssl)
   # This is required to access OpenSSL compatibility API
   target_include_directories(jwt-cpp INTERFACE ${wolfssl_INCLUDE_DIRS})
-  target_compile_definitions(jwt-cpp INTERFACE OPENSSL_EXTRA OPENSSL_ALL)
+  # EXTERNAL_OPTS_OPENVPN is necessary so that wolfssl/options.h is included automatically
```

Comment:
Why was this header not required before? Would it be possible to just at the header?

Reply:
It was always necessary to include `wolfssl/options.h`. Notice that in your current builds you have
```
In file included from /usr/local/include/wolfssl/openssl/bn.h:33,
                 from /usr/local/include/wolfssl/openssl/ec.h:27,
                 from /home/runner/work/jwt-cpp/jwt-cpp/include/jwt-cpp/jwt.h:15,
                 from /home/runner/work/jwt-cpp/jwt-cpp/tests/JwksTest.cpp:1:
/usr/local/include/wolfssl/wolfcrypt/settings.h:2369:14: warning: #warning "For timing resistance / side-channel attack prevention consider using harden options" [-Wcpp]
 2369 |             #warning "For timing resistance / side-channel attack prevention consider using harden options"
      |              ^~~~~~~
```
which is a dead giveaway that `wolfssl/options.h` is missing. I think by defining `OPENSSL_EXTRA OPENSSL_ALL` you were able to somewhat build with wolfSSL but I don't know how the tests are "passing". I didn't investigate this.